docs-developer/custom-importer.md

```diff
@@ -0,0 +1,76 @@
+
+```js
+{
+  // Get the first sample out of the first thread:
+  const threadIndex = 0;
+  const sampleIndex = 0;
+
+  const thread = profile.threads[threadIndex];
+
+  // Look up the stack index from the samples table.
+  const stackIndex = thread.samples.stack[sampleIndex];
+  console.log({stackIndex})
+
+  // Look up the frame from the stack table.
+  const frameIndex = thread.stackTable.frame[stackIndex];
+  console.log({frameIndex})
+
+  // Look up the function from the frame table.
+  const funcIndex = thread.frameTable.func[frameIndex];
+  console.log({funcIndex})
+
+  // Look up the string index from the func table.
+  const stringIndex = thread.funcTable.name[funcIndex];
+  console.log({stringIndex})
+
+  console.log(
+    'Function name of the first sample:',
+    thread.stringTable.getString(stringIndex)
+  );
+}
+```
+
+It is probably to read up some on profile format docs for more information on how this structure is used.
+
+## Tips
+
+ * Use `{ name: 'GeckoMain' }` for the main thread.
+ * Ensure that the `pid` value is numeric, and points to the proper threads to nest threads in the timeline.
+ * Processed profiles have their timestamps adjusted so that all processes use the same timeline.
+ * Gecko profiles adjust the marker and sample start times based on when the process was created (based off of `geckoProfile.meta.startTime`).
+
+## Samples
+
+The call tree, flame graph, and stack chart all use the samples data to generate the reports. If your data is shaped like sample data, it can be easy to shove the information in there. At the time of this writing we don't support variable durations. It's assumed 1 samples is equal in length to the sampling interval. This is probably going to change soon. Most likely the profile formats will assume that there is at least one sample in a thread. It might be necessary to fill out the samples with dummy time data if your format isn't needing samples. The sample data should match the time range as the marker data.
+
+## Markers
+
+Markers are used in the marker chart, marker table, and for various visualization in the header's timeline. For instance they are drawn as solid squares. Screenshots are encoded as [ScreenshotPayload](https://github.com/firefox-devtools/profiler/search?q=ScreenshotPayload&unscoped_q=ScreenshotPayload) and extracted as a separate track. Markers can be used to encode some tracing types of data, as they can have start and end times.
+
+One important distinction in the UI is that there is some processing going on everytime you view a profile. The importer should target the RawMarkerTable, while the UI surfaces the `filteredMarkers` on the window object. The filtered markers do some processing to match up start and end markers. This is a bit beyond the scope of this document, but the type definitions should explain things a bit more. It could be a source of confusion when writing and working with markers.
```

Comment:
Maybe link to the type for `RawMarkerTable` ?

Is it worth explaining quickly the 3 types of markers ?
1. markers without a duration (they can be anything)
2. pairs of tracing markers (they have `type === 'tracing'`)
3. "full" marker with a duration (they have `startTime`/`endTime` in their payload)